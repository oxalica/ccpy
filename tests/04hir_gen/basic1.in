1 + 2
pass
(2, 3)
f((), x, -y.z, (None,))
__intrinsic__wtf()      # Unrecognize. Normal call.
__intrinsic__id1(42)    # Intrinsic call.
__intrinsic__id1        # Not call.
f(1)(2)(3)
1 and 2 and 3
1 or 2 and 3
not 1
not not 1
1 if 2 else 3
