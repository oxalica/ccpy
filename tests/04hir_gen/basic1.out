Start(Local size = 8):
0: %0 = 1
1: %1 = 2
2: %2 = __intrinsic__get_global0()
3: %3 = "__builtin__add"
4: %4 = None
5: %2 = __intrinsic__dict_get3(%2, %3, %4)
6: %0 = __intrinsic__tuple_make_(%0, %1)
7: %0 = __intrinsic__v_call2(%2, %0)
8: %0 = 2
9: %1 = 3
10: %0 = __intrinsic__tuple_make_(%0, %1)
11: %0 = "f"
12: %1 = __intrinsic__get_global0()
13: %2 = "__builtin__global_get"
14: %3 = None
15: %1 = __intrinsic__dict_get3(%1, %2, %3)
16: %0 = __intrinsic__tuple_make_(%0)
17: %0 = __intrinsic__v_call2(%1, %0)
18: %1 = __intrinsic__tuple_make_()
19: %2 = "x"
20: %3 = __intrinsic__get_global0()
21: %4 = "__builtin__global_get"
22: %5 = None
23: %3 = __intrinsic__dict_get3(%3, %4, %5)
24: %2 = __intrinsic__tuple_make_(%2)
25: %2 = __intrinsic__v_call2(%3, %2)
26: %3 = "y"
27: %4 = __intrinsic__get_global0()
28: %5 = "__builtin__global_get"
29: %6 = None
30: %4 = __intrinsic__dict_get3(%4, %5, %6)
31: %3 = __intrinsic__tuple_make_(%3)
32: %3 = __intrinsic__v_call2(%4, %3)
33: %4 = "z"
34: %5 = __intrinsic__get_global0()
35: %6 = "__builtin__getattr"
36: %7 = None
37: %5 = __intrinsic__dict_get3(%5, %6, %7)
38: %3 = __intrinsic__tuple_make_(%3, %4)
39: %3 = __intrinsic__v_call2(%5, %3)
40: %4 = __intrinsic__get_global0()
41: %5 = "__builtin__neg"
42: %6 = None
43: %4 = __intrinsic__dict_get3(%4, %5, %6)
44: %3 = __intrinsic__tuple_make_(%3)
45: %3 = __intrinsic__v_call2(%4, %3)
46: %4 = None
47: %4 = __intrinsic__tuple_make_(%4)
48: %1 = __intrinsic__tuple_make_(%1, %2, %3, %4)
49: %2 = __intrinsic__get_global0()
50: %3 = "__builtin__call"
51: %4 = None
52: %2 = __intrinsic__dict_get3(%2, %3, %4)
53: %0 = __intrinsic__tuple_make_(%0, %1)
54: %0 = __intrinsic__v_call2(%2, %0)
55: %0 = "__intrinsic__wtf"
56: %1 = __intrinsic__get_global0()
57: %2 = "__builtin__global_get"
58: %3 = None
59: %1 = __intrinsic__dict_get3(%1, %2, %3)
60: %0 = __intrinsic__tuple_make_(%0)
61: %0 = __intrinsic__v_call2(%1, %0)
62: %1 = __intrinsic__tuple_make_()
63: %2 = __intrinsic__get_global0()
64: %3 = "__builtin__call"
65: %4 = None
66: %2 = __intrinsic__dict_get3(%2, %3, %4)
67: %0 = __intrinsic__tuple_make_(%0, %1)
68: %0 = __intrinsic__v_call2(%2, %0)
69: %0 = 42
70: %0 = __intrinsic__id1(%0)
71: %0 = "__intrinsic__id1"
72: %1 = __intrinsic__get_global0()
73: %2 = "__builtin__global_get"
74: %3 = None
75: %1 = __intrinsic__dict_get3(%1, %2, %3)
76: %0 = __intrinsic__tuple_make_(%0)
77: %0 = __intrinsic__v_call2(%1, %0)
78: %0 = "f"
79: %1 = __intrinsic__get_global0()
80: %2 = "__builtin__global_get"
81: %3 = None
82: %1 = __intrinsic__dict_get3(%1, %2, %3)
83: %0 = __intrinsic__tuple_make_(%0)
84: %0 = __intrinsic__v_call2(%1, %0)
85: %1 = 1
86: %1 = __intrinsic__tuple_make_(%1)
87: %2 = __intrinsic__get_global0()
88: %3 = "__builtin__call"
89: %4 = None
90: %2 = __intrinsic__dict_get3(%2, %3, %4)
91: %0 = __intrinsic__tuple_make_(%0, %1)
92: %0 = __intrinsic__v_call2(%2, %0)
93: %1 = 2
94: %1 = __intrinsic__tuple_make_(%1)
95: %2 = __intrinsic__get_global0()
96: %3 = "__builtin__call"
97: %4 = None
98: %2 = __intrinsic__dict_get3(%2, %3, %4)
99: %0 = __intrinsic__tuple_make_(%0, %1)
100: %0 = __intrinsic__v_call2(%2, %0)
101: %1 = 3
102: %1 = __intrinsic__tuple_make_(%1)
103: %2 = __intrinsic__get_global0()
104: %3 = "__builtin__call"
105: %4 = None
106: %2 = __intrinsic__dict_get3(%2, %3, %4)
107: %0 = __intrinsic__tuple_make_(%0, %1)
108: %0 = __intrinsic__v_call2(%2, %0)
109: %0 = 1
110: %1 = __intrinsic__get_global0()
111: %2 = "__builtin__not"
112: %3 = None
113: %1 = __intrinsic__dict_get3(%1, %2, %3)
114: %2 = __intrinsic__tuple_make_(%0)
115: %1 = __intrinsic__v_call2(%1, %2)
116: JmpIfFalse %1 <120>
117: %2 = %0
118: %0 = False
119: JmpIfFalse %0 <134>
120: %3 = 2
121: %4 = __intrinsic__get_global0()
122: %5 = "__builtin__not"
123: %6 = None
124: %4 = __intrinsic__dict_get3(%4, %5, %6)
125: %5 = __intrinsic__tuple_make_(%3)
126: %4 = __intrinsic__v_call2(%4, %5)
127: JmpIfFalse %4 <131>
128: %5 = %3
129: %3 = False
130: JmpIfFalse %3 <133>
131: %6 = 3
132: %5 = %6
133: %2 = %5
134: %0 = 1
135: %1 = __intrinsic__get_global0()
136: %2 = "__builtin__not"
137: %3 = None
138: %1 = __intrinsic__dict_get3(%1, %2, %3)
139: %2 = __intrinsic__tuple_make_(%0)
140: %1 = __intrinsic__v_call2(%1, %2)
141: JmpIfFalse %1 <158>
142: %3 = 2
143: %4 = __intrinsic__get_global0()
144: %5 = "__builtin__not"
145: %6 = None
146: %4 = __intrinsic__dict_get3(%4, %5, %6)
147: %5 = __intrinsic__tuple_make_(%3)
148: %4 = __intrinsic__v_call2(%4, %5)
149: JmpIfFalse %4 <153>
150: %5 = %3
151: %3 = False
152: JmpIfFalse %3 <155>
153: %6 = 3
154: %5 = %6
155: %2 = %5
156: %3 = False
157: JmpIfFalse %3 <159>
158: %2 = %0
159: %0 = 1
160: %1 = __intrinsic__get_global0()
161: %2 = "__builtin__not"
162: %3 = None
163: %1 = __intrinsic__dict_get3(%1, %2, %3)
164: %0 = __intrinsic__tuple_make_(%0)
165: %0 = __intrinsic__v_call2(%1, %0)
166: %0 = 1
167: %1 = __intrinsic__get_global0()
168: %2 = "__builtin__not"
169: %3 = None
170: %1 = __intrinsic__dict_get3(%1, %2, %3)
171: %0 = __intrinsic__tuple_make_(%0)
172: %0 = __intrinsic__v_call2(%1, %0)
173: %1 = __intrinsic__get_global0()
174: %2 = "__builtin__not"
175: %3 = None
176: %1 = __intrinsic__dict_get3(%1, %2, %3)
177: %0 = __intrinsic__tuple_make_(%0)
178: %0 = __intrinsic__v_call2(%1, %0)
179: %0 = 2
180: %1 = __intrinsic__get_global0()
181: %2 = "__builtin__not"
182: %3 = None
183: %1 = __intrinsic__dict_get3(%1, %2, %3)
184: %0 = __intrinsic__tuple_make_(%0)
185: %0 = __intrinsic__v_call2(%1, %0)
186: JmpIfFalse %0 <191>
187: %2 = 3
188: %1 = %2
189: %2 = False
190: JmpIfFalse %2 <193>
191: %3 = 1
192: %1 = %3
193: %0 = 1
194: %1 = __intrinsic__get_global0()
195: %2 = "__builtin__not"
196: %3 = None
197: %1 = __intrinsic__dict_get3(%1, %2, %3)
198: %0 = __intrinsic__tuple_make_(%0)
199: %0 = __intrinsic__v_call2(%1, %0)
200: JmpIfFalse %0 <215>
201: %1 = 3
202: %2 = __intrinsic__get_global0()
203: %3 = "__builtin__not"
204: %4 = None
205: %2 = __intrinsic__dict_get3(%2, %3, %4)
206: %1 = __intrinsic__tuple_make_(%1)
207: %1 = __intrinsic__v_call2(%2, %1)
208: JmpIfFalse %1 <212>
209: %2 = 5
210: %2 = False
211: JmpIfFalse %2 <213>
212: %3 = 4
213: %1 = False
214: JmpIfFalse %1 <216>
215: %2 = 2


